Index: src/view/new-point.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/view/new-point.js b/src/view/new-point.js
--- a/src/view/new-point.js	(revision 0d4e2da49c791b52ff474e04015bb3c3f4e0eda7)
+++ b/src/view/new-point.js	(date 1638300453168)
@@ -1,138 +1,145 @@
-const createPictureTemplate = (arr) => `<div class="event__photos-container">
+const createPictureTemplate = (
+  pictures
+) => `<div class="event__photos-container">
     <div class="event__photos-tape">
-      ${arr.map((src) => `<img class="event__photo" src="${src}" alt="Event photo"></img>`).join('')}
+      ${pictures
+    .map(
+      ({ src }) =>
+        `<img class="event__photo" src='${src}' alt="Event photo"/>`
+    )
+    .join('')}
     </div>
   </div>`;
 
-const createOfferTemplate = (arr) => `<section class="event__section  event__section--offers">
+
+const createOfferTemplate = (
+  offers
+) => `<section class="event__section  event__section--offers">
     <h3 class="event__section-title  event__section-title--offers">Offers</h3>
     <div class="event__available-offers">
-        ${arr.map(({id, title, price}) => `<div class="event__offer-selector">
+        ${offers
+    .map(
+      ({ id, title, price }) => `<div class="event__offer-selector">
           <input class="event__offer-checkbox  visually-hidden" id="${id}-1" type="checkbox" name="${id}" checked>
           <label class="event__offer-label" for="${id}-1">
             <span class="event__offer-title">${title}</span>
             &plus;&euro;&nbsp;
             <span class="event__offer-price">${price}</span>
           </label>
-        </div>`).join('')}
+        </div>`
+    )
+    .join('')}
       </div>
   </section>`;
 
-export const createNewPointTemplate = (point = {}) => {
-  const {
-    type = 'Flight',
-    price = '',
-    destination = {
-      name: 'Geneva',
-      description: 'Geneva is a city in Switzerland that lies at the southern tip of expansive Lac Léman (Lake Geneva). Surrounded by the Alps and Jura mountains, the city has views of dramatic Mont Blanc.',
-      pictures: [],
-    },
-    offer = [],
-  } = point;
+export const createNewPointTemplate = (point) => {
+  const { destination, offers } = point;
 
-  const offerTemplate = offer.length ? createOfferTemplate(offer) : '';
-  const pictureTemplate = destination.pictures ? createPictureTemplate(destination.pictures) : '';
+  const offerTemplate = offers.length ? createOfferTemplate(offers) : '';
+  const pictureTemplate = destination.pictures
+    ? createPictureTemplate(destination.pictures)
+    : '';
 
-  `<form class="event event--edit" action="#" method="post">
-                <header class="event__header">
-                  <div class="event__type-wrapper">
-                    <label class="event__type  event__type-btn" for="event-type-toggle-1">
-                      <span class="visually-hidden">Choose event type</span>
-                      <img class="event__type-icon" width="17" height="17" src="img/icons/bus.png" alt="Event type icon">
-                    </label>
-                    <input class="event__type-toggle  visually-hidden" id="event-type-toggle-1" type="checkbox">
+  return `<form class="event event--edit" action="#" method="post">
+    <header class="event__header">
+      <div class="event__type-wrapper">
+        <label class="event__type  event__type-btn" for="event-type-toggle-1">
+          <span class="visually-hidden">Choose event type</span>
+          <img class="event__type-icon" width="17" height="17" src="img/icons/bus.png" alt="Event type icon">
+        </label>
+        <input class="event__type-toggle  visually-hidden" id="event-type-toggle-1" type="checkbox">
 
-                    <div class="event__type-list">
-                      <fieldset class="event__type-group">
-                        <legend class="visually-hidden">Event type</legend>
+        <div class="event__type-list">
+          <fieldset class="event__type-group">
+            <legend class="visually-hidden">Event type</legend>
 
-                        <div class="event__type-item">
-                          <input id="event-type-taxi-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="taxi">
-                          <label class="event__type-label  event__type-label--taxi" for="event-type-taxi-1">Taxi</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-taxi-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="taxi">
+              <label class="event__type-label  event__type-label--taxi" for="event-type-taxi-1">Taxi</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-bus-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="bus" checked>
-                          <label class="event__type-label  event__type-label--bus" for="event-type-bus-1">Bus</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-bus-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="bus" checked>
+              <label class="event__type-label  event__type-label--bus" for="event-type-bus-1">Bus</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-train-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="train">
-                          <label class="event__type-label  event__type-label--train" for="event-type-train-1">Train</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-train-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="train">
+              <label class="event__type-label  event__type-label--train" for="event-type-train-1">Train</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-ship-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="ship">
-                          <label class="event__type-label  event__type-label--ship" for="event-type-ship-1">Ship</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-ship-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="ship">
+              <label class="event__type-label  event__type-label--ship" for="event-type-ship-1">Ship</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-drive-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="drive">
-                          <label class="event__type-label  event__type-label--drive" for="event-type-drive-1">Drive</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-drive-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="drive">
+              <label class="event__type-label  event__type-label--drive" for="event-type-drive-1">Drive</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-flight-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="flight">
-                          <label class="event__type-label  event__type-label--flight" for="event-type-flight-1">Flight</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-flight-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="flight">
+              <label class="event__type-label  event__type-label--flight" for="event-type-flight-1">Flight</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-check-in-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="check-in">
-                          <label class="event__type-label  event__type-label--check-in" for="event-type-check-in-1">Check-in</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-check-in-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="check-in">
+              <label class="event__type-label  event__type-label--check-in" for="event-type-check-in-1">Check-in</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-sightseeing-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="sightseeing">
-                          <label class="event__type-label  event__type-label--sightseeing" for="event-type-sightseeing-1">Sightseeing</label>
-                        </div>
+            <div class="event__type-item">
+              <input id="event-type-sightseeing-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="sightseeing">
+              <label class="event__type-label  event__type-label--sightseeing" for="event-type-sightseeing-1">Sightseeing</label>
+            </div>
 
-                        <div class="event__type-item">
-                          <input id="event-type-restaurant-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="restaurant">
-                          <label class="event__type-label  event__type-label--restaurant" for="event-type-restaurant-1">Restaurant</label>
-                        </div>
-                      </fieldset>
-                    </div>
-                  </div>
+            <div class="event__type-item">
+              <input id="event-type-restaurant-1" class="event__type-input  visually-hidden" type="radio" name="event-type" value="restaurant">
+              <label class="event__type-label  event__type-label--restaurant" for="event-type-restaurant-1">Restaurant</label>
+            </div>
+          </fieldset>
+        </div>
+      </div>
 
-                  <div class="event__field-group  event__field-group--destination">
-                    <label class="event__label  event__type-output" for="event-destination-1">
-                      Bus
-                    </label>
-                    <input class="event__input  event__input--destination" id="event-destination-1" type="text" name="event-destination" value="{destination.name}" list="destination-list-1">
-                    <datalist id="destination-list-1">
-                      <option value="Amsterdam"></option>
-                      <option value="Geneva"></option>
-                      <option value="Chamonix"></option>
-                    </datalist>
-                  </div>
+      <div class="event__field-group  event__field-group--destination">
+        <label class="event__label  event__type-output" for="event-destination-1">
+          Bus
+        </label>
+        <input class="event__input  event__input--destination" id="event-destination-1" type="text" name="event-destination" value="${destination.name}" list="destination-list-1">
+        <datalist id="destination-list-1">
+          <option value="Amsterdam"></option>
+          <option value="Geneva"></option>
+          <option value="Chamonix"></option>
+        </datalist>
+      </div>
 
-                  <div class="event__field-group  event__field-group--time">
-                    <label class="visually-hidden" for="event-start-time-1">From</label>
-                    <input class="event__input  event__input--time" id="event-start-time-1" type="text" name="event-start-time" value="19/03/19 00:00">
-                    &mdash;
-                    <label class="visually-hidden" for="event-end-time-1">To</label>
-                    <input class="event__input  event__input--time" id="event-end-time-1" type="text" name="event-end-time" value="19/03/19 00:00">
-                  </div>
+      <div class="event__field-group  event__field-group--time">
+        <label class="visually-hidden" for="event-start-time-1">From</label>
+        <input class="event__input  event__input--time" id="event-start-time-1" type="text" name="event-start-time" value="19/03/19 00:00">
+        &mdash;
+        <label class="visually-hidden" for="event-end-time-1">To</label>
+        <input class="event__input  event__input--time" id="event-end-time-1" type="text" name="event-end-time" value="19/03/19 00:00">
+      </div>
 
-                  <div class="event__field-group  event__field-group--price">
-                    <label class="event__label" for="event-price-1">
-                      <span class="visually-hidden">Price</span>
-                      &euro;
-                    </label>
-                    <input class="event__input  event__input--price" id="event-price-1" type="text" name="event-price" value="">
-                  </div>
+      <div class="event__field-group  event__field-group--price">
+        <label class="event__label" for="event-price-1">
+          <span class="visually-hidden">Price</span>
+          &euro;
+        </label>
+        <input class="event__input  event__input--price" id="event-price-1" type="text" name="event-price" value="">
+      </div>
 
-                  <button class="event__save-btn  btn  btn--blue" type="submit">Save</button>
-                  <button class="event__reset-btn" type="reset">Cancel</button>
-                </header>
+      <button class="event__save-btn  btn  btn--blue" type="submit">Save</button>
+      <button class="event__reset-btn" type="reset">Cancel</button>
+    </header>
 
-                <section class="event__details">
-                  ${offerTemplate}
-                  <section class="event__section  event__section--destination">
-                    <h3 class="event__section-title  event__section-title--destination">Destination</h3>
-                    <p class="event__destination-description">${destination.description}</p>
-                     ${pictureTemplate}
-                  </section>
-                </section>
-              </form>`;
+    <section class="event__details">
+      ${offerTemplate}
+      <section class="event__section  event__section--destination">
+        <h3 class="event__section-title  event__section-title--destination">Destination</h3>
+        <p class="event__destination-description">${destination.description}</p>
+         ${pictureTemplate}
+      </section>
+    </section>
+  </form>`;
 };
Index: src/view/point-view.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/view/point-view.js b/src/view/point-view.js
--- a/src/view/point-view.js	(revision 0d4e2da49c791b52ff474e04015bb3c3f4e0eda7)
+++ b/src/view/point-view.js	(date 1638300245519)
@@ -1,7 +1,5 @@
-import flatpickr from 'flatpickr';
-
-const createOfferTemplate = (arr) => `<ul class="event__selected-offers">
-  ${arr.map(({title, price}) => `<li class="event__offer">
+const createOfferTemplate = (offers) => `<ul class="event__selected-offers">
+  ${offers.map(({title, price}) => `<li class="event__offer">
     <span class="event__offer-title">${title}</span>
     &plus;&euro;&nbsp;
     <span class="event__offer-price">${price}</span>
@@ -9,45 +7,38 @@
 </ul>`;
 
 export const createPointTemplate = (point) => {
-  const {destination, pointType, offer, dateCheckIn, dateCheckOut, price} = point;
+  const {destination, pointType, offers, dateCheckIn, dateCheckOut, price} = point;
 
   const favoriteClassName = point.isFavorite ? 'event__favorite-btn--active' : '';
-  const offerTemplate = offer.length ? createOfferTemplate(offer) : '';
+  const offerTemplate = offers.length ? createOfferTemplate(offers) : '';
 
   return `<div class="event">
-                <time class="event__date" datetime="2019-03-18">MAR 18</time>
-                <div class="event__type">
-                  <img class="event__type-icon" width="42" height="42" src="img/icons/taxi.png" alt="Event type icon">
-                </div>
-                <h3 class="event__title">${destination + pointType}</h3>
-                <div class="event__schedule">
-                  <p class="event__time">
-                    <time class="event__start-time" datetime="2019-03-18T10:30">${dateCheckIn}</time>
-                    &mdash;
-                    <time class="event__end-time" datetime="2019-03-18T11:00">${dateCheckOut}</time>
-                  </p>
-                  <p class="event__duration">30M</p>
-                </div>
-                <p class="event__price">
-                  &euro;&nbsp;<span class="event__price-value">${price}</span>
-                </p>
-                <h4 class="visually-hidden">Offers:</h4>
-                  ${offerTemplate}
-                // <ul class="event__selected-offers">
-                //   <li class="event__offer">
-                //     <span class="event__offer-title">Order Uber</span>
-                //     &plus;&euro;&nbsp;
-                //     <span class="event__offer-price">${price}</span>
-                //   </li>
-                // </ul>
-                <button class="event__favorite-btn ${favoriteClassName}" type="button">
-                  <span class="visually-hidden">Add to favorite</span>
-                  <svg class="event__favorite-icon" width="28" height="28" viewBox="0 0 28 28">
-                    <path d="M14 21l-8.22899 4.3262 1.57159-9.1631L.685209 9.67376 9.8855 8.33688 14 0l4.1145 8.33688 9.2003 1.33688-6.6574 6.48934 1.5716 9.1631L14 21z"/>
-                  </svg>
-                </button>
-                <button class="event__rollup-btn" type="button">
-                  <span class="visually-hidden">Open event</span>
-                </button>
-              </div>`;
+    <time class="event__date" datetime="2019-03-18">MAR 18</time>
+    <div class="event__type">
+      <img class="event__type-icon" width="42" height="42" src="img/icons/taxi.png" alt="Event type icon">
+    </div>
+    <h3 class="event__title">${pointType} ${destination.name}</h3>
+    <div class="event__schedule">
+      <p class="event__time">
+        <time class="event__start-time" datetime="2019-03-18T10:30">${dateCheckIn}</time>
+        &mdash;
+        <time class="event__end-time" datetime="2019-03-18T11:00">${dateCheckOut}</time>
+      </p>
+      <p class="event__duration">30M</p>
+    </div>
+    <p class="event__price">
+      &euro;&nbsp;<span class="event__price-value">${price}</span>
+    </p>
+    <h4 class="visually-hidden">Offers:</h4>
+      ${offerTemplate}
+    <button class="event__favorite-btn ${favoriteClassName}" type="button">
+      <span class="visually-hidden">Add to favorite</span>
+      <svg class="event__favorite-icon" width="28" height="28" viewBox="0 0 28 28">
+        <path d="M14 21l-8.22899 4.3262 1.57159-9.1631L.685209 9.67376 9.8855 8.33688 14 0l4.1145 8.33688 9.2003 1.33688-6.6574 6.48934 1.5716 9.1631L14 21z"/>
+      </svg>
+    </button>
+    <button class="event__rollup-btn" type="button">
+      <span class="visually-hidden">Open event</span>
+    </button>
+  </div>`;
 };
Index: package-lock.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package-lock.json b/package-lock.json
--- a/package-lock.json	(revision 0d4e2da49c791b52ff474e04015bb3c3f4e0eda7)
+++ b/package-lock.json	(date 1638300447356)
@@ -1342,6 +1342,11 @@
         "which": "^2.0.1"
       }
     },
+    "dayjs": {
+      "version": "1.10.7",
+      "resolved": "https://registry.npmjs.org/dayjs/-/dayjs-1.10.7.tgz",
+      "integrity": "sha512-P6twpd70BcPK34K26uJ1KT3wlhpuOAPoMwJzpsIWUxHZ7wpmbdZL/hQqBDfz7hGurYSa5PhzdhDHtt319hL3ig=="
+    },
     "debug": {
       "version": "4.3.2",
       "resolved": "https://registry.npmjs.org/debug/-/debug-4.3.2.tgz",
Index: src/main.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main.js b/src/main.js
--- a/src/main.js	(revision 0d4e2da49c791b52ff474e04015bb3c3f4e0eda7)
+++ b/src/main.js	(date 1638298569632)
@@ -29,7 +29,7 @@
 
 const eventsElementList = mainElement.querySelector('.trip-events__list');
 
-renderTemplate(eventsElementList, createPointsListItemTemplate(createNewPointTemplate()), RenderPosition.BEFOREEND);
+renderTemplate(eventsElementList, createPointsListItemTemplate(createNewPointTemplate(points[0])), RenderPosition.BEFOREEND);
 for (let i = 0; i < POINTS_COUNT; i++) {
   renderTemplate(eventsElementList, createPointsListItemTemplate(createPointTemplate(points[i])), RenderPosition.BEFOREEND);
 }
Index: src/mock/task.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/mock/task.js b/src/mock/task.js
--- a/src/mock/task.js	(revision 0d4e2da49c791b52ff474e04015bb3c3f4e0eda7)
+++ b/src/mock/task.js	(date 1638300263080)
@@ -1,5 +1,4 @@
-import flatpickr from 'flatpickr';
-
+import dayjs from 'dayjs';
 // Функция из интернета по генерации случайного числа из диапазона
 // Источник - https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_random
 const getRandomInteger = (a = 0, b = 1) => {
@@ -9,17 +8,7 @@
   return Math.floor(lower + Math.random() * (upper - lower + 1));
 };
 
-const generateDestination = () => {
-  const destinations = [
-    'Amsterdam',
-    'Chamonix',
-    'Geneva',
-  ];
-  const randomIndex = getRandomInteger(0, destinations.length - 1);
-  return destinations[randomIndex];
-};
-
-const generatePointTipe = () => {
+const generatePointType = () => {
   const pointTypes = [
     'Taxi',
     'Bus',
@@ -35,33 +24,50 @@
   return pointTypes[randomIndex];
 };
 
-const generateTravelDateCheckIn = () => {
-  const randomTimeCheckIn = getRandomInteger(1, 31);
-  const config = {
-    enableTime: true,
-    dateFormat: 'Y-m-d H:i',
-  };
-  return flatpickr(randomTimeCheckIn, config);
-};
-
-const generateTravelDateCheckOut = () => {
-  const randomTimeCheckOut = getRandomInteger(1, 31);
-  // if (randomTimeCheckOut < randomTimeCheckIn) {
-  //   return null;
-  const config = {
-    enableTime: true,
-    dateFormat: 'Y-m-d H:i',
-  };
-  return flatpickr(randomTimeCheckOut, config);
-  // }
-};
-
 const priceGeneration = () => getRandomInteger(0, 1000);
 
+
+const generateOffers = () => {
+  const titles = ['Upgrade to a business class', 'Choose the radio station'];
+
+  return {
+    type: generatePointType(),
+    offers: Array.from({ length: getRandomInteger(2, 4) }, (_, i) => ({
+      id: i,
+      title: titles[i],
+      price: priceGeneration()
+    }))
+  };
+};
+
+const generatePictures = () => Array.from({ length: getRandomInteger(1, 3) }, () => ({
+  src: `http://picsum.photos/248/152?r=${getRandomInteger(1, 5)}`,
+  description: 'Chamonix parliament building'
+}));
+
+const generateDestination = () => {
+  const destinations = [
+    'Amsterdam',
+    'Chamonix',
+    'Geneva',
+  ];
+  return {
+    description: 'Chamonix, is a beautiful city, a true asian pearl, with crowded streets.',
+    name: destinations[getRandomInteger(0, 2)],
+    pictures: generatePictures()
+  };
+};
+
+const generateTravelDateCheckIn = () => dayjs(new Date()).format('DD MMMM');
+
+const generateTravelDateCheckOut = () => dayjs(new Date()).add(4, 'day').format('DD MMMM');
+
 export const generatePoint = () => ({
   destination: generateDestination(),
-  pointType: generatePointTipe(),
+  pointType: generatePointType(),
   dateCheckIn: generateTravelDateCheckIn(),
   dateCheckOut: generateTravelDateCheckOut(),
   price: priceGeneration(),
+  offers: generateOffers(),
+  isFavorite: Boolean(getRandomInteger(0, 1)),
 });
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision 0d4e2da49c791b52ff474e04015bb3c3f4e0eda7)
+++ b/package.json	(date 1638300447283)
@@ -26,6 +26,7 @@
     "node": "16"
   },
   "dependencies": {
+    "dayjs": "1.10.7",
     "flatpickr": "^4.6.9"
   }
 }
